io/io/src/TFile.cxx

```diff
@@ -867,7 +867,7 @@ void TFile::Init(Bool_t create)
             ReadStreamerInfo();
             if (IsZombie()) {
                R__LOCKGUARD(gROOTMutex);
-               gROOT->GetListOfFiles()->Remove(this);
+               if (addToList) gROOT->GetListOfFiles()->Remove(this);
```

Comment:
The line 869 is now redundant (and in the spirit of this PR would have needed to inside the if clause), as GetListOfFiles is now a thread safe container.